contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -516,11 +523,15 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     s_dons[id].id = id;
     s_dons[id].isPublic = isPublic;
 
+    ++s_dons[id].configCount;
+    uint32 configCount = s_dons[id].configCount;
+
     for (uint256 i; i < nodes.length; ++i) {
       bytes32 nodeP2PId = nodes[i];
-      if (s_dons[id].nodes.contains(nodeP2PId)) revert DuplicateDONNode(id, nodeP2PId);
+      if (s_dons[id].nodeCapabilityConfigs[configCount].nodes.contains(nodeP2PId))
+        revert DuplicateDONNode(id, nodeP2PId);
```

Comment:
We could probably do this validation in memory?

Reply:
I don't think we can load this into memory as we have a nested mapping